cpp/scripts/run-clang-format.py

```diff
@@ -22,6 +22,7 @@
 import tempfile
 
 
+EXPECTED_VERSION = "8.0.1"
```

Comment:
This is restricting clang to version `8.0.1` but the definitions above allow for any version of `8.X.Y`. Shouldn't we relax this or at least fix to `8.0.1`?

Reply:
I think we should pin to 8.0.1 completely. We've seen past clang releases make subtle changes to formatting rules that cause PRs to start failing the style checks. So we only want to update to a new minor version if we've tested it manually in advance.

Reply:
Changed to pin conda envs to 8.0.1